File: java/calculator.c
#include <stdio.h>

int main() {
    char operator;
    double num1, num2, result;

    // Get input from user
    printf("Enter first number: ");
    scanf("%lf", &num1);
    
    printf("Enter an operator (+, -, *, /, %%): ");
    scanf(" %c", &operator);  // Note the space before %c to consume any whitespace
    
    printf("Enter second number: ");
    scanf("%lf", &num2);

    // Perform calculation using switch case
    switch(operator) {
        case '+':
            result = num1 + num2;
            printf("%.2lf + %.2lf = %.2lf\n", num1, num2, result);
            break;
            
        case '-':
            result = num1 - num2;
            printf("%.2lf - %.2lf = %.2lf\n", num1, num2, result);
            break;
            
        case '*':
            result = num1 * num2;
            printf("%.2lf * %.2lf = %.2lf\n", num1, num2, result);
            break;
            
        case '/':
            if(num2 != 0) {
                result = num1 / num2;
                printf("%.2lf / %.2lf = %.2lf\n", num1, num2, result);
            } else {
                printf("Error! Division by zero is not allowed.\n");
            }
            break;
            
        case '%':
            if(num2 != 0) {
                result = (int)num1 % (int)num2;  // Type casting to int for modulus
                printf("%d %% %d = %d\n", (int)num1, (int)num2, (int)result);
            } else {
                printf("Error! Division by zero is not allowed.\n");
            }
            break;
            
        default:
            printf("Error! Invalid operator. Please enter +, -, *, /, or %%\n");
    }

    return 0;
} 

File: java/Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean running = true;

        System.out.println("Welcome to Basic Calculator!");
        System.out.println("Available operations: +, -, *, /");
        System.out.println("Enter 'q' to quit");

        while (running) {
            System.out.print("\nEnter first number: ");
            String input1 = scanner.next();

            if (input1.equalsIgnoreCase("q")) {
                running = false;
                continue;
            }

            double num1;
            try {
                num1 = Double.parseDouble(input1);
            } catch (NumberFormatException e) {
                System.out.println("Invalid input! Please enter a number.");
                continue;
            }

            System.out.print("Enter operation (+, -, *, /): ");
            String operation = scanner.next();

            if (operation.equalsIgnoreCase("q")) {
                running = false;
                continue;
            }

            if (!operation.equals("+") && !operation.equals("-") &&
                    !operation.equals("*") && !operation.equals("/")) {
                System.out.println("Invalid operation! Please use +, -, *, or /");
                continue;
            }

            System.out.print("Enter second number: ");
            String input2 = scanner.next();

            if (input2.equalsIgnoreCase("q")) {
                running = false;
                continue;
            }

            double num2;
            try {
                num2 = Double.parseDouble(input2);
            } catch (NumberFormatException e) {
                System.out.println("Invalid input! Please enter a number.");
                continue;
            }

            double result = 0;
            boolean validOperation = true;

            if (operation.equals("+")) {
                result = num1 + num2;
            } else if (operation.equals("-")) {
                result = num1 - num2;
            } else if (operation.equals("*")) {
                result = num1 * num2;
            } else if (operation.equals("/")) {
                if (num2 == 0) {
                    System.out.println("Error: Division by zero!");
                    validOperation = false;
                } else {
                    result = num1 / num2;
                }
            }

            if (validOperation) {
                System.out.printf("Result: %.2f %s %.2f = %.2f%n", num1, operation, num2, result);
            }
        }

        System.out.println("Thank you for using the calculator!");
        scanner.close();
    }
}